doc.go

```diff
@@ -18,121 +18,235 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-// Package dig is the dig: Dependency Injection Framework for Go.
-//
-// package dig provides an opinionated way of resolving object dependencies.
+// Package dig provides an opinionated way of resolving object dependencies.
 //
 // Status
 //
-// BETA. Expect potential API changes.
+// STABLE. No breaking changes will be made in this major version.
 //
 // Container
 //
-// package dig exposes type Container as an object capable of resolving a
-// directional dependency graph.
+// Dig exposes type Container as an object capable of resolving a directional
+// dependency graph. To create one, simply use the New function.
 //
+//   c := dig.New()
 //
-// To create one:
+// Provide
 //
-//   import "go.uber.org/dig"
+// Constructors for different types are added to the container by using the
+// Provide method. A constructor can declare a dependency on another type by
+// simply adding it as a function parameter. Dependencies for a type can be
+// added to the graph both, before and after that type was added.
 //
-//   func main() {
-//   	c := dig.New()
-//   	// dig container `c` is ready to use!
+//   err := c.Provide(func(conn *sql.DB) (*UserGateway, error) {
+//     // ...
+//   })
+//   if err != nil {
+//     // ...
 //   }
 //
-// **All objects in the container are treated as a singletons**, meaning there can be
-// only one object in the graph of a given type.
+//   if err := c.Provide(newDBConnection); err != nil {
+//     // ...
+//   }
 //
+// Multiple constructors can rely on the same type. Types in the container are
```

Comment:
The container creates a singleton for each retained type. (solves passive voice by making the container the subject of the sentence)